From 7ad8ccde39d7c258ad596279ba9cb5f9339a4409 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Wed, 25 Mar 2015 15:04:15 +0100
Subject: [PATCH] Remove crash while cross-compiling in VPATH mode

The dummy config.h file created by native update-icon-cache should be
saved in $(builddir), not in $(srcdir).
---
 gtk/native/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gtk/native/Makefile.am b/gtk/native/Makefile.am
index 7516dab..ef1fc4c 100644
--- a/gtk/native/Makefile.am
+++ b/gtk/native/Makefile.am
@@ -5,9 +5,9 @@ AM_CPPFLAGS = @CPPFLAGS_FOR_BUILD@
 AM_LDFLAGS = @LDFLAGS_FOR_BUILD@
 
 native-update-icon-cache$(BUILD_EXEEXT): $(srcdir)/../updateiconcache.c
-	echo '#define GETTEXT_PACKAGE "gtk30"' > $(srcdir)/config.h
-	$(AM_V_CCLD)$(CC_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) -I. -I$(srcdir)/../.. $(NATIVE_GDKPIXBUF_CFLAGS) $(CFLAGS_FOR_BUILD) $^ $(LDFLAGS_FOR_BUILD) $(NATIVE_GDKPIXBUF_LIBS) $(LIBS_FOR_BUILD) -o $@
-	rm -f $(srcdir)/config.h
+	echo '#define GETTEXT_PACKAGE "gtk30"' > $(builddir)/config.h
+	$(AM_V_CCLD)/usr/bin/env -i PATH="/usr/bin:$$PATH" $(CC_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) -I$(builddir) $(NATIVE_GDKPIXBUF_CFLAGS) $(CFLAGS_FOR_BUILD) $^ $(LDFLAGS_FOR_BUILD) $(NATIVE_GDKPIXBUF_LIBS) $(LIBS_FOR_BUILD) -o $@
+	rm -f $(builddir)/config.h
 
 if CROSS_COMPILING
 if !USE_EXTERNAL_ICON_CACHE
-- 
2.3.3

